index.js

```diff
@@ -99,30 +99,35 @@ UglifyWriter.prototype.processFile = function(inFile, outFile, relativePath, out
   var src = fs.readFileSync(inFile, 'utf-8');
   var mapName = path.basename(outFile).replace(/\.js$/,'') + '.map';
   var mapDir;
-  var origSourcesContent;
 
   if (this.sourceMapConfig.mapDir) {
     mapDir = path.join(outDir, this.sourceMapConfig.mapDir);
   } else {
     mapDir = path.dirname(path.join(outDir, relativePath));
   }
 
+  var sourceMapOptions = {
+    includeSources: this.options.sourceMapIncludeSources,
+    filename: inFile,
+    url: this.mapURL(mapName)
+  };
+
   var opts = {
-    fromString: true,
-    outSourceMap: this.mapURL(mapName),
-    enableSourcemaps: this.enableSourcemaps()
+    compress: this.options.compress,
+    mangle: this.options.mangle,
+    sourceMap: this.enableSourcemaps() ? sourceMapOptions : false
```

Comment:
ah this makes stuff a tad tricky.

I suspect the first step will be a major version bump of this library. But we would also like to include it in new ember-cli blueprints. So we may need to explore potential upgrade pains/and come up with a gentle upgrade plan if it is too painful.

Thoughts?